packages/core/src/index.ts

```diff
@@ -8,15 +8,22 @@ import {
   KB_JWT_TYP,
   SDJWTCompact,
   SDJWTConfig,
-  SD_JWT_TYP,
 } from '@sd-jwt/types';
 
 export * from './sdjwt';
 export * from './kbjwt';
 export * from './jwt';
 export * from './decoy';
 
-export class SDJwtInstance {
+export interface SDJwtPayload {
+  // more entries
+  [key: string]: unknown;
+}
+
+export abstract class SDJwtInstance<ExtendedPayload extends SDJwtPayload> {
+  //header type
+  protected abstract type: string;
```

Comment:
Can we set a default `type` is 'sd-jwt'? @cre8

Reply:
@lukasjhan no, because there is no specified format with only `sd-jwt` so the library forces you to implement the class.
The `sd-jwt` only defines the algorithm that is used inside a JWT, it says nothing about the other components like header attributes or signature formats